api/src/main/java/org/apache/iceberg/RewriteFiles.java

```diff
@@ -34,13 +34,57 @@
  * will throw a {@link ValidationException}.
  */
 public interface RewriteFiles extends SnapshotUpdate<RewriteFiles> {
+  /**
+   * Delete a data file whose content was rewritten.
```

Comment:
Using the word `delete` is a bit questionable in this context but that's what the Javadoc says. It is also similar to the naming used in `OverwriteFiles` and other APIs. I can see this being called `rewriteFile` or something too.

Reply:
Our terminology always throw me a bit off here. We struggle with "delete" = physically remove and "delete" mark as logically removed. Both are used in various places and sometimes we also use "remove".

Here I think "originalFile" maybe is more of the logical description, "original" => "rewritten" instead of "add" and "delete" which are really the implementation details. 

That said i'm fine with just "add" and "delete" since we use them in many other places.

Reply:
Yeah, it is unfortunate we overlooked the distinction between physical `delete` and logical `remove`. Let me double check other APIs to see whether there are also any good examples.

Reply:
Does anyone else have any preferences?

Reply:
@jackye1995 @rdblue, any thoughts?